### PR TITLE
[doc/example] Make LDAP example functional again by running OpenLDAP with docker-compose

### LINKED ISSUE #1654: missing ./scripts/slapd.sh
I would follow the [LDAP tutorial](https://github.com/dexidp/dex/blob/master/Documentation/connectors/ldap.md) but the script `./scripts/slapd.sh` is missing.

### PR DESCRIPTION
**Pull request by heidemn-faro**
Fixes #1654

### CODE REVIEW

File: examples/config-ldap.yaml [hunk 1]
```diff
@@ -11,7 +11,7 @@ connectors:
   name: OpenLDAP
   id: ldap
   config:
-    host: localhost:10389
+    host: localhost:389
```
Comment (heidemn-faro) [inline examples/config-ldap.yaml:14]: If there's a strong opinion to use the nonstandard port, I can revert it.
But then the port would be different here (outside the container) and when using "docker exec" (inside the container).

**Comment by sagikazarmark** [comment]
Thanks @heidemn-faro . Personally, I'd much rather see a docker-compose based solution, without adding back that script. I believe the OpenLDAP image is able to add definitions automatically on startup.

**Comment by heidemn-faro** [comment]
@sagikazarmark okay understand.
But since the issue is open since February, I would suggest that you accept this as a temporary fix until you have time to adapt it according to your preferences.

At least it works now if you follow the docs :-)

**Comment by heidemn-faro** [comment]
@sagikazarmark as requested, I've reworked this to use docker-compose.
Any further changes required?

File: examples/config-ldap.yaml (hunk 1 above)
Comment (sagikazarmark) [reply examples/config-ldap.yaml:14]: I think the rationale behind using a non-standard port is avoiding potential collision with an existing service. But all the other examples (mysql, etc) use standard ports, so it should be okay.

**Comment by heidemn-faro** [comment]
@sagikazarmark  @bonifaido thanks for reviewing.
The PR is ready from my side. I've verified that the example works.

I don't have permissions to merge, so you would have to do that.

**Pull request merged**